docs/source/tutorials/memory_optimisations.rst

```diff
@@ -0,0 +1,243 @@
+.. _memory_optimisation_overview_label:
+
+============================
+Memory Optimization Overview
+============================
+
+**Author**: `Salman Mohammadi <https://github.com/SalmanMohammadi>`_
+
+torchtune comes with a host of plug-and-play memory optimization components which give you lots of flexibility
+to ``tune`` our recipes to your hardware. This page provides a brief glossary of these components and how you might use them.
```

Comment:
Commenting at the top after having read this whole tutorial: I think this is a great start and a sorely-needed tutorial. Just a couple things to think about to give it a more unified feel (doesn't have to be part of this PR, this is more aspirational). 

Right now this leans more towards an index of techniques with some tradeoff discussions, I'd like to see it become more of a cookbook of what to apply and when. To that end, we could shoot for some kind of table enumerating (a) various memory-savings techniques and their impact, (b) implications for performance and/or model quality, and (c) guidance on which ones to try or combine under different regimes. Anyways this is a longer-term thing so no need to worry about it for this PR, would definitely be interested to hear your thoughts on whether something like that makes sense to you.

Reply:
As discussed offline, both your's and Rafi's suggestions are on-the-nose. 

Can we start collating these ideas centrally? We have this, #954, #1163, #1260, #1252, #1282